buildSrc/src/main/java/org/opensearch/gradle/test/DistroTestPlugin.java

```diff
@@ -77,9 +77,9 @@
 import java.util.stream.Stream;
 
 public class DistroTestPlugin implements Plugin<Project> {
-    private static final String SYSTEM_JDK_VERSION = "21.0.4+7";
+    private static final String SYSTEM_JDK_VERSION = "23+37";
```

Comment:
For 2.x, I would change it to `21.0.4+7` since 23 has some known regressions, no need to risk 2.18.0